modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -425,8 +426,11 @@ const DomParser = (settings: DomParserSettings = {}, schema = Schema()): DomPars
     const body = parser.parseFromString(wrappedHtml, mimeType).body;
 
     // Sanitize the content
-    purify.sanitize(body, getPurifyConfig(defaultedSettings, mimeType));
-    purify.removed = [];
+    if (defaultedSettings.sanitize) {
+      const purify = setupPurify(defaultedSettings, schema);
```

Comment:
I'm not sure about moving the setup code in here. I guess if it causes performance issues we can deal with it later.

Reply:
Maybe it could be pulled out to a function like `runDomPurify` or something would make it easier to test that in isolation if we wanted to.

Reply:
Agreed, I should've seen that the setupPurify function is now being run every single time content is parsed instead of just when the DomParser is defined like before.